crates/rustc_codegen_spirv/src/codegen_cx/constant.rs

```diff
@@ -530,8 +534,9 @@ impl<'tcx> CodegenCx<'tcx> {
         read_target_uint(self.data_layout().endian, bytes).unwrap()
     }
 
-    // Advances offset by ptr size
-    fn read_alloc_ptr<'a>(&self, alloc: &'a Allocation, offset: &mut Size) -> Pointer {
+    // Advances offset by ptr size, None indicates that the Scalar did not have relocations
+    // and that it was non zero
+    fn read_alloc_ptr<'a>(&self, alloc: &'a Allocation, offset: &mut Size) -> Option<Scalar> {
```

Comment:
Just so it doesn't get forgotten, we've discussed on Discord about replacing these two functions with `Allocation::read_scalar` from rustc itself.